synthetic_data/dataset_generator.py

```diff
@@ -0,0 +1,124 @@
+"""Contains generator that returns collective df of requested distinct generators."""
+
+from datetime import datetime
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from numpy.random import Generator
+
+from synthetic_data.distinct_generators.categorical_generator import random_categorical
+from synthetic_data.distinct_generators.datetime_generator import random_datetimes
+from synthetic_data.distinct_generators.float_generator import random_floats
+from synthetic_data.distinct_generators.int_generator import random_integers
+from synthetic_data.distinct_generators.text_generator import random_string, random_text
+
+
+def convert_data_to_df(
+    np_data: np.array,
+    path: Optional[str] = None,
+    index: bool = False,
+    column_names: Optional[List[str]] = None,
+) -> pd.DataFrame:
+    """
+    Convert np array to a pandas dataframe.
+
+    :param np_data: np array to be converted
+    :type np_data: numpy array
+    :param path: path to output a csv of the dataframe generated
+    :type path: str, None, optional
+    :param index: whether to include index in output to csv
+    :type path: bool, optional
+    :param column_names: The names of the columns of a dataset
+    :type path: List, None, optional
+    :return: a pandas dataframe
+    """
+    # convert array into dataframe
+    if not column_names:
+        column_names = [x for x in range(len(np_data))]
+    dataframe = pd.DataFrame.from_dict(dict(zip(column_names, np_data)))
+
+    # save the dataframe as a csv file
+    if path:
+        dataframe.to_csv(path, index=index, encoding="utf-8")
+    return dataframe
+
+
+def get_ordered_column(
+    data: np.array,
+    original_format: str = "%B %d %Y %H:%M:%S", # tab gen will take 
+    order: str = "ascending",
+) -> np.array:
+    """Sort a numpy array based on data type.
+
+    :param data: numpy array to be sorted
+    :type data: np.array
+
+    :return: sorted numpy array
+    """
+    if type == "datetime":
+        date_object = np.array([datetime.strptime(dt, original_format) for dt in data])
+        sorted_datetime = np.sort(date_object)
+        sorted_data = np.array([dt.strftime(original_format) for dt in sorted_datetime])
+    else:
+        sorted_data = np.sort(data)
+    if order == "descending":
+        return sorted_data[::-1]
+    return sorted_data
```

Comment:
We should avoid creating a datetime here that sin't using our generator. I would suggest possibly not even allowing datetime cols to get here and ensure they adhere to the order inside the datetime generators themselves. This will require restructuring that generator unfortunately.

The problem is a col can have multiple formats and this would fail in that casel.

I suggest updating the datetime generator to possibly take a different approach where possibly inside based on `num_rows` it gets an array for `t` the len of `num_rows` and sorts it. Subsequently, it does the date \ format generation. This would allow for sorting inside if desired.

i'm not keen on it, but I think it is problematic here to try to convert back and do extra work at this point.

Reply:
that limits this just to the sort with alteration if descending.

Reply:
so datetime_generator will always return a sorted datetime array?